### PR TITLE
gh-1080 Rest - Fix Creating a Stream causes a `500` error

### PR DESCRIPTION
**Pull request by ghillert**
- Add new exception classes
  - `InvalidStreamDefinitionException`
  - `CannotDetermineApplicationTypeException`
- Ensure a `400` Http status error is returned
- Add Tests

Resolves https://github.com/spring-cloud/spring-cloud-dataflow/issues/1080

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/InvalidStreamDefinitionException.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.controller.support;
+
+/**
+ *
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/InvalidStreamDefinitionException.java]: Maybe add some text here.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/package-info.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/**
+ *
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/support/package-info.java]: Maybe add some text here.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/support/CannotDetermineApplicationTypeException.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.dataflow.server.support;
+
+/**
+ *
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/support/CannotDetermineApplicationTypeException.java]: Maybe add some text here.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/support/package-info.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/**
+ *
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/support/package-info.java]: Maybe add some text here.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RestControllerAdvice.java [hunk 5]
```diff
@@ -135,7 +136,10 @@ public VndErrors onNotFoundException(Exception e) {
 	 * Log the exception message at warn level and stack trace as trace level.
 	 * Return response status HttpStatus.BAD_REQUEST (400).
 	 */
-	@ExceptionHandler({MissingServletRequestParameterException.class})
+	@ExceptionHandler({
+		MissingServletRequestParameterException.class,
```
Comment (jvalkeal) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RestControllerAdvice.java]: I think I had a merge conflict here as you probably didn't PR from latest master.

**Comment by jvalkeal** [comment]
@ghillert Do you plan to modify this PR?

**Comment by ghillert** [comment]
@jvalkeal Updated PR with some JavaDoc

**Comment by jvalkeal** [comment]
Closing, merge/squashed per https://github.com/spring-cloud/spring-cloud-dataflow/commit/b16174ee9c93df74bed6cacd72865a4c00fee6d1.

**Pull request closed without merging**